lib/galaxy/datatypes/speech.py

```diff
@@ -0,0 +1,120 @@
+from galaxy.datatypes.text import Text
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.metadata import MetadataElement, ListParameter, DictParameter
+from galaxy import util
+
+import wave
```

Comment:
this is an additional package we will need to ship. I wonder if we maybe should use the `ffprobe` function from the media datatypes to reduce dependencies (then only needing ffmpeg to be installed). It looks like it reads nicely as a file with a single audio stream.

```
  "streams": [
    {
      "DISPOSITION:attached_pic": "0",
      "DISPOSITION:clean_effects": "0",
      "DISPOSITION:comment": "0",
      "DISPOSITION:default": "0",
      "DISPOSITION:dub": "0",
      "DISPOSITION:forced": "0",
      "DISPOSITION:hearing_impaired": "0",
      "DISPOSITION:karaoke": "0",
      "DISPOSITION:lyrics": "0",
      "DISPOSITION:original": "0",
      "DISPOSITION:visual_impaired": "0",
      "avg_frame_rate": "0/0",
      "bit_rate": "128000",
      "bits_per_raw_sample": "N/A",
      "bits_per_sample": "16",
      "channel_layout": "unknown",
      "channels": "1",
      "codec_long_name": "PCM signed 16-bit little-endian",
      "codec_name": "pcm_s16le",
      "codec_tag": "0x0001",
      "codec_tag_string": "[1][0][0][0]",
      "codec_time_base": "1/8000",
      "codec_type": "audio",
      "duration": "0.615375",
      "duration_ts": "4923",
      "id": "N/A",
      "index": "0",
      "max_bit_rate": "N/A",
      "nb_frames": "N/A",
      "nb_read_frames": "N/A",
      "nb_read_packets": "N/A",
      "profile": "unknown",
      "r_frame_rate": "0/0",
      "sample_fmt": "s16",
      "sample_rate": "8000",
      "start_pts": "N/A",
      "start_time": "N/A",
      "time_base": "1/8000"
    }
  ]
```

Reply:
wave is part of the standard library, so not an extra dependency.  ffprobe would be able to understand more file formats (eg. mp3) and it pulls out quite a bit more metadata that might be of interest (eg. I noticed it has an encoded_by field for the software that generated the file which would be great in metadata)

So, if we're going to have a dependency on ffprobe then we should use it for audio too.

Reply:
Huh! I didn't know wave was part of stdlib, that's really nifty!